lib/elixir/lib/integer.ex

```diff
@@ -225,12 +254,12 @@ defmodule Integer do
 
   """
   @spec to_string(integer, 2..36) :: String.t
-  def to_string(some_integer, base) do
-    :erlang.integer_to_binary(some_integer, base)
+  def to_string(integer, base) when is_integer(integer) and is_integer(base) and base in 2..36 do
```

Comment:
Is there any point in adding guards to a function that is inlined by the compiler?